src/main/java/org/datadog/jenkins/plugins/datadog/apm/GoConfigurator.java

```diff
@@ -0,0 +1,176 @@
+package org.datadog.jenkins.plugins.datadog.apm;
+
+import hudson.FilePath;
+import hudson.model.Node;
+import hudson.model.TaskListener;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.datadog.jenkins.plugins.datadog.clients.HttpClient;
+import org.datadog.jenkins.plugins.datadog.steps.TestOptimization;
+import org.datadog.jenkins.plugins.datadog.util.SuppressFBWarnings;
+
+public class GoConfigurator implements TracerConfigurator {
+    private static final Logger LOGGER = Logger.getLogger(GoConfigurator.class.getName());
+
+    private static final String TRACER_VERSION_ENV_VAR = "DD_SET_TRACER_VERSION_GO";
+    private static final String LATEST_TAG = "latest";
+    private static final String ORCHESTRION_LATEST_URL = "https://api.github.com/repos/datadog/orchestrion/releases/latest";
```

Comment:
I wonder if we can come up with a fallback for cases when the current shell user is not logged in to GitHub